primitives/src/xdns.rs

```diff
@@ -33,6 +33,46 @@ pub struct FetchXdnsRecordsResponse<AccountId> {
     pub xdns_records: Vec<XdnsRecord<AccountId>>,
 }
 
+/// A preliminary representation of a xdns_record in the onchain registry.
+#[derive(Clone, Encode, Decode, Eq, PartialEq, Debug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct TokenRecord {
+    /// Its token 4b Id
+    pub token_id: [u8; 4],
+
+    /// Link to the gateway token is whitelisted on.
+    pub gateway_id: [u8; 4],
+
+    /// Token properties - decimals, symbol, name
+    pub token_props: TokenSysProps,
```

Comment:
How does any of this differ to pallet-assets/registry? Is this not just doing the same thing

Reply:
Very possibly 👍

Reply:
what would u suggest then, XDNS with the access to pallet-assets or vice versa? 

Circuit should verify if asset is registered at SFX validation before submission. For now Circuit access XDNS

Reply:
I would suggest that the circuit should query pallet-assets for existence of our version of that asset, XDNS could also map to the asset index if it **really** wants to, but everything should live in assets/asset-registry